cypher-manual/clauses/call.md

```diff
@@ -0,0 +1,253 @@
+    1.1. [Cartesian products](#11-cartesian-products) <br />
+    1.2. [Cartesian products with bounded symbols](#12-cartesian-products-with-bounded-symbols) <br />
+    1.3. [Post-union processing](#13-post-union-processing) <br />
+    1.4. [Observing changes from previous executions](#14-observing-changes-from-previous-executions)<br />
+    1.5. [Unit subqueries](#15-unit-subqueries)
+
+2. [Invalid uses of CALL subquery](#2-invalid-uses-of-call-subquery) <br />
+    2.1. [Returning variables with the same name as those in the outer scope](#21-returning-variables-with-same-name-as-those-in-the-outer-scope) <br />
+    2.2. [Returning non aliased expressions](#22-returning-non-aliased-expressions) <br />
+    2.3. [Referencing outer scope variables that do not exist](#22-referencing-outer-scope-variables-that-do-not-exist) <br />
+
+# 1. Uses of CALL subquery
+## 1.1. Cartesian products
+
+`CALL` subquery is executed once for each incoming row. If there are multiple rows that get produced from the
+`CALL` subquery, the result is a Cartesian product of results. You can think of this as combining the output of
+2 branches, one being called the `input branch` (the produced rows before calling the subquery), and the `subquery branch`
+(the produced rows of the subquery).
+
+```cypher
+MATCH (p:Person)
+CALL {
+  MATCH (a:Animal)
+  RETURN a.name as animal_name
+}
+RETURN p.name as person_name, animal_name
+```
+
+If we assume that there are 2 `:Person` in the database named `John` and `Alice`, 
+and 2 `:Animal` named `Rex` and `Lassie`, the result would be as follows:
+
+Output:
+```nocopy
++-------------+-------------+
+| person_name | animal_name |
++---------------------------+
+| 'John'      | 'Rex'       |
+| 'John'      | 'Lassie'    |
+| 'Alice'     | 'Rex'       |
+| 'Alice'     | 'Lassie'    |
++---------------------------+
+```
+
+## 1.2. Cartesian products with bounded symbols
+
+By referencing variables in the subquery from the outer scope using the `WITH` keyword,
+we can use those same symbols to expand on the neighborhood of the referenced nodes or relationships.
+By not using the `WITH` keyword as the first clause in the query, we will not have the access to the
+outer scope variables, and the subquery would behave as it sees the variable for the first time.
+
+```cypher
+MATCH (person:Person)
+CALL {
+  WITH person
+  MATCH (person)-[:HAS_PARENT]->(parent:Parent)
+  RETURN parent
+}
+RETURN person.name, parent.name
+```
+
+Output:
+```nocopy
++-------------+-------------+
+| person_name | animal_name |
++---------------------------+
+| 'John'      | 'John Sr.'  |
+| 'John'      | 'Anna'      |
+| 'Alice'     | 'Roxanne'   |
+| 'Alice'     | 'Bill'      |
++---------------------------+
+```
+
+## 1.3. Post-union processing
+
+When doing unions, users can combine output from all the union queries inside a subquery, in order to
+post processing on their data and make the query more expressive.
```

Comment:
```suggestion
Output from all UNION queries inside a subquery can be combined and forwarded as a single output to make the queries more expressive:
```

Reply:
Is seems like i'm missing something with: in order to post processing?

Reply:
it's good like this as well